test/k8sT/Policies.go

```diff
@@ -827,78 +736,157 @@ EOF`, k, v)
+			_ = kubectl.Delete(netpolNsSelector)
+
 		})
 
 		AfterAll(func() {
-			By("Deleting the namespace that was used for the pods")
-			err := kubectl.CoreV1().Namespaces().Delete(namespace, &metav1.DeleteOptions{})
-			Expect(err).NotTo(HaveOccurred(), "cannot delete %q namespace", namespace)
+			_ = kubectl.Delete(demoManifest)
+			_ = kubectl.NamespaceDelete(namespace)
 		})
 
-		It("should enforce policy based on NamespaceSelector", func() {
-			By("Testing pods can connect to both ports when no policy is present.")
-			testCanConnect(kubectl, namespace, "client-can-connect-80", service, 80, true)
-			testCanConnect(kubectl, namespace, "client-can-connect-81", service, 81, true)
-
-			By("Apply the network policy")
-			policyBeforeCreate := &networkingv1.NetworkPolicy{
-				ObjectMeta: metav1.ObjectMeta{
-					Name: policy,
-				},
-				Spec: networkingv1.NetworkPolicySpec{
-					// Apply to server
-					PodSelector: metav1.LabelSelector{
-						MatchLabels: map[string]string{
-							"pod-name": podServer.Name,
-						},
-					},
-					// Allow traffic only from NS-B
-					Ingress: []networkingv1.NetworkPolicyIngressRule{{
-						From: []networkingv1.NetworkPolicyPeer{{
-							NamespaceSelector: &metav1.LabelSelector{
-								MatchLabels: map[string]string{
-									"ns-name": namespace,
-								},
-							},
-						}},
-					}},
-				},
+		It("Tests the same Policy in the different namespaces", func() {
+
+			By("Applying Policy in %q namespace", namespace)
+			_, err = kubectl.CiliumPolicyAction(
+				helpers.KubeSystemNamespace, policy, helpers.KubectlApply, 300)
+			Expect(err).Should(BeNil(), "L3 Policy cannot be applied in %q namespace", namespace)
+
+			By("Applying Policy in default namespace")
+			_, err = kubectl.CiliumPolicyAction(
+				helpers.KubeSystemNamespace, l3Policy, helpers.KubectlApply, 300)
+			Expect(err).Should(BeNil(), "L3 Policy cannot be applied in %q namespace", helpers.DefaultNamespace)
+
+			By("Testing %q namespace", namespace)
+
+			res := kubectl.ExecPodCmd(
+				namespace, appPodsNS[helpers.App2],
+				helpers.CurlFail(fmt.Sprintf("http://%s/public", clusterIPNS)))
+			res.ExpectSuccess("%q cannot curl service", appPods[helpers.App2])
+
+			res = kubectl.ExecPodCmd(
+				namespace, appPodsNS[helpers.App3],
+				helpers.CurlFail(fmt.Sprintf("http://%s/public", clusterIPNS)))
+			res.ExpectFail("%q can curl to service", appPods[helpers.App3])
+
+			By("Testing default namespace")
+
+			res = kubectl.ExecPodCmd(
+				helpers.DefaultNamespace, appPods[helpers.App2],
+				helpers.CurlFail(fmt.Sprintf("http://%s/public", clusterIP)))
+			res.ExpectSuccess("%q cannot curl clusterIP %q", appPods[helpers.App2], clusterIP)
+
+			res = kubectl.ExecPodCmd(
+				helpers.DefaultNamespace, appPods[helpers.App3],
+				helpers.CurlFail(fmt.Sprintf("http://%s/public", clusterIP)))
+			res.ExpectFail("%q can curl to %q", appPods[helpers.App3], clusterIP)
+		})
+
+		It("Network Policy namespace selector", func() {
```

Comment:
This is not clear enough. What is this testing? Make the string reflect what this tests overall.